UncertainSCI/utils/compute_subintervals.py

```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+def compute_subintervals(a, b, singularity_list):
+    """
+    Returns an M x 4 numpy array, where each row contains the left-hand point,
+    right-hand point, left-singularity strength, and right-singularity
+    strength.
+    """
+
+    # Tolerance for resolving internal versus boundary singularities.
+    tol = 1e-12
+    singularities  = np.array([entry[0] for entry in singularity_list])
+    strength_left  = np.array([entry[1] for entry in singularity_list])
+    strength_right = np.array([entry[2] for entry in singularity_list])
+
+    # We can discard any singularities that lie to the left of a or the right of b
+    discard = []
+    for (ind,s) in enumerate(singularities):
+        if s < a-tol or s > b+tol:
+            discard.append(ind)
+
+    singularities  = np.delete(singularities, discard)
```

Comment:
This code could use a refactoring--the same multi-step procedure is done multiple times to 3 different arrays. That smells like a couple methods/classes needing to be extracted.